### ISSUE TITLE
Datastore conflict AutoMerge behaves differently from amplify-android

### ISSUE THREAD

**Issue by HuiSF** [aws-amplify/amplify-swift#1308]
### Describe the bug

Original issue amplify-flutter [#634](https://github.com/aws-amplify/amplify-flutter/issues/634)

amplify-ios AutoMerge for handling conflict caused by multi-device offline/online syncing behaves differently from amplify-android, where amplify-android seems having the correct behavior.

Detailed description please refer to [this comment](https://github.com/aws-amplify/amplify-flutter/issues/634#issuecomment-873151792)

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. Run datastore in two iOS simulators, say device A and device B, both have same data
2. Make device A and B both offline
3. Make changes to a piece of data in device A, quit App, then make device A online, reopen the App to publish the update mutation
4. Make changes to the same piece of data in device B, quit App, then make device B online, reopen the App to publish the update mutation
```

### Expected behavior

The update mutation made in device B when it was offline should be merged into DynamoDB after AutoMerge

### Amplify Framework Version

1.11.0

### Amplify Categories

DataStore

### Dependency manager

Cocoapods

### Swift version

5

### CLI version

5.1.0

### Xcode version

12.5.1

### Device

iPhone 12 simulators

### iOS Version

iOS 14.5

**Comment by chrisbonifacio** [comment]
@HuiSF sorry the delay, we've triaged the issue and added it to our backlog for improving behavior parity between platforms.

**Comment by atierian** [comment]
This fix has been released in [2.25.3](https://github.com/aws-amplify/amplify-swift/releases/tag/2.25.3) and [1.30.6](https://github.com/aws-amplify/amplify-swift/releases/tag/1.30.6)

Thanks for reporting this and for your patience!
